hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/metadata/SparkHoodieBackedTableMetadataWriter.java

```diff
@@ -118,46 +123,32 @@ protected void initRegistry() {
   }
 
   @Override
-  protected <T extends SpecificRecordBase> void initialize(HoodieEngineContext engineContext,
-                                                           Option<T> actionMetadata,
-                                                           Option<String> inflightInstantTimestamp) {
-    try {
-      metrics.map(HoodieMetadataMetrics::registry).ifPresent(registry -> {
-        if (registry instanceof DistributedRegistry) {
-          HoodieSparkEngineContext sparkEngineContext = (HoodieSparkEngineContext) engineContext;
-          ((DistributedRegistry) registry).register(sparkEngineContext.getJavaSparkContext());
-        }
-      });
+  protected void commit(String instantTime, Map<MetadataPartitionType, HoodieData<HoodieRecord>> partitionRecordsMap) {
+    commitInternal(instantTime, partitionRecordsMap, Option.empty());
+  }
 
-      if (enabled) {
-        initializeIfNeeded(dataMetaClient, actionMetadata, inflightInstantTimestamp);
-      }
-    } catch (IOException e) {
-      LOG.error("Failed to initialize metadata table. Disabling the writer.", e);
-      enabled = false;
-    }
+  protected void bulkCommit(
+          String instantTime, MetadataPartitionType partitionType, HoodieData<HoodieRecord> records,
+          int fileGroupCount) {
+    Map<MetadataPartitionType, HoodieData<HoodieRecord>> partitionRecordsMap = new HashMap<>();
+    partitionRecordsMap.put(partitionType, records);
+    SparkHoodieMetadataBulkInsertPartitioner partitioner = new SparkHoodieMetadataBulkInsertPartitioner(fileGroupCount);
```

Comment:
Can bulk_insert supporting writing HFile directly ? It seems it still write parquets, then the k-v query performance should not be good ?

Reply:
It will write HFile since the base file format of the MDT is HFile.

Reply:
You are right, thanks for the clarification.